api/views.py

```diff
@@ -0,0 +1,94 @@
+    def __init__(self):
+        """constructor method for the class"""
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('client', type=str, required=True,\
+         help='please provide your name')
+        self.reqparse.add_argument('contact', type=int, required=True,\
+         help='please provide your contact')
+        self.reqparse.add_argument('order_item', type=str, required=True,\
+         help='please provide your order')
+        self.reqparse.add_argument('price', type=int, required=True,\
+         help='please enter the price')
+
+    def get(self):
+        """
+        method to fetch all orders
+        """
+        response = jsonify({'orders': orders})
+        if len(orders) == 0:
+            return jsonify({'message': 'No orders have been placed yet'})
+        return response
+    
+    def post(self):
+        """
+        method to post a question
+        """
+        order_id = len(orders) + 1
+        order_status = 'submitted'
+        args = self.reqparse.parse_args()
+        response = Orders(order_id, args['client'], args['contact'], args['order_item'],\
+         args['price'], order_status)
+        response = response.to_json()
+        if response:
+            client_name = args['client']
+            client_name = client_name.strip()
+            if not re.match(r"^[a-zA-Z]+$", client_name):
+                return make_response(jsonify({'message': 'Username should only have letters'}), 400)
+            orders.append(response)
+            return make_response(jsonify({'message': 'Order has been placed'}), 201)
+        return make_response(jsonify({'message': 'Order not placed'}), 400)
+
+class SpecificOrder(Resource):
+    """
+    class handles methods for specific order
+    """
+    def __init__(self):
+        """
+        constructor method for the class
+        """
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('order_status', type=str, required=True,\
+         help='please provide your name')
+
+    def get(self, order_id):
+        """
+        method to get a specific question
+        :param order_id:
+        :return: order
+        """
+        for order in orders:
+            if order['order_id'] == order_id:
+                return order
+        return make_response(jsonify({'message': 'order not found'}), 404)
+    
+    def put(self, order_id):
+        """
+        method to update order-status
+        :param order_id:
+        """
+        for order in orders:
+            if order['order_id'] == order_id:
+                args = self.reqparse.parse_args()
+                response = args['order_status']
+                if response:
+                    order['order_status'] = response
+                    return make_response(jsonify({'message': 'order status updated'}), 201)
+                return make_response(jsonify({'message': 'order status not updated'}), 400)
```

Comment:
`400` is  a more generic error code, use a more specific status code, it would also be better to tell the user why the status was not updated (`The why`) instead of the `The what`

Reply:
I am using 400 because, the user input is invalid for my my application.
Refer to this resource https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
I don't if it changes your opinion about my usage of  400.

Reply:
>400 Bad Request
This response means that server could not understand the request due to invalid syntax.

It doesn't change my opinion because the syntax is valid, If you are telling the user that you were unable to process the data because he gave you a wrong format, then I would go for a `422` `Unprocessed Entity`